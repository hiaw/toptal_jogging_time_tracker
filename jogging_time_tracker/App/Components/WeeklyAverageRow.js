/* @flow*/
import React from 'react'
import { Text, View } from 'react-native'

import { getDistanceText, getSpeedText } from '../Helper/SpeedCalculator.js'

import styles from './Styles/WeeklyAverageRow.style.js'

export type Props = {
  distance: number,
  duration: number,
}

const WeeklyAverageRow = props => {
  const { distance, duration } = props
  const distanceText = getDistanceText(distance)
  const speedText = getSpeedText(distance, duration)

  return (
    <View style={styles.container}>
      <Text style={styles.welcome}>
        Total: {distanceText}
      </Text>
      <Text style={styles.welcome}>
        Avg: {speedText}
      </Text>
    </View>
  )
}

export default WeeklyAverageRow
