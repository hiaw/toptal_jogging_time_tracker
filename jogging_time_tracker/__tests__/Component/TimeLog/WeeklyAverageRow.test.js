import 'react-native'
import React from 'react'

import WeeklyAverageRow from '../../../App/Components/TimeLog/WeeklyAverageRow.js'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

const noop = () => {}

it('renders correctly', () => {
  const tree = renderer
    .create(<WeeklyAverageRow distance={34.3} duration={34.3} />)
    .toJSON()
  expect(tree).toMatchSnapshot()
})
