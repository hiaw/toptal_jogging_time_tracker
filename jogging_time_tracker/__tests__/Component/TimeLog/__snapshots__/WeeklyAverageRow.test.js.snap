// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "aliceblue",
      "borderBottomColor": "black",
      "borderBottomWidth": 0.5,
      "flexDirection": "row",
      "justifyContent": "space-between",
      "paddingHorizontal": 20,
      "paddingVertical": 10,
    }
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Total: 
    34.3 m
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
      }
    }
  >
    Avg: 
    3.6 km/h
  </Text>
</View>
`;
